### PR TITLE
Improve work with tooltip in deckGL

### PR DESCRIPTION
**Pull request by vdubr**
Nasazuju dokumentaci kde je použití.

Bude to vidět u Vector vrstvy a i WMS vrstvy

### CODE REVIEW

**Comment by vlach1989** [comment]
I'd add some kind of restriction, if there are too many features overlapping

File: src/DeckGlMap/index.js [hunk 1]
```diff
@@ -29,13 +30,59 @@ const DeckGlMap = ({
...
+	const onHover = useCallback(
+		event => {
+			if (deckRef?.current?.pickMultipleObjects) {
+				const hoveredItems = deckRef?.current?.pickMultipleObjects(event);
+				const vectorHoveredItems = [];
+				const rasterHoveredItems = [];
+				hoveredItems.forEach(item => {
+					if (item.layer instanceof VectorLayer) {
+						//add to vector items
+						console.log('xxx_isvector');
```
Comment (vlach1989) [inline src/DeckGlMap/index.js:56]: remove log

File: src/DeckGlMap/index.js [hunk 2]
```diff
@@ -29,13 +30,59 @@ const DeckGlMap = ({
...
+						vectorHoveredItems.push(item);
+					} else if (item.layer instanceof WmsLayer) {
+						console.log('xxx_israster');
```
Comment (vlach1989) [inline src/DeckGlMap/index.js:59]: remove log

File: src/DeckGlMap/index.js [hunk 3]
```diff
@@ -29,13 +30,59 @@ const DeckGlMap = ({
...
+						//this is path fot tile layer
+						const image =
+							item?.tile?.layers?.[0]?.props?.tile?.layers?.[0]?.props?.image;
+						const pixelColor = readPixelsToArray(image, {
+							sourceX: event?.bitmap?.pixel?.[0],
+							sourceY: event?.bitmap?.pixel?.[1],
+							sourceWidth: 1,
+							sourceHeight: 1,
+						});
+						item.pixelColor = pixelColor;
```
Comment (vlach1989) [inline src/DeckGlMap/index.js:70]:
```suggestion
						item.pixelColor = readPixelsToArray(image, {
							sourceX: event?.bitmap?.pixel?.[0],
							sourceY: event?.bitmap?.pixel?.[1],
							sourceWidth: 1,
							sourceHeight: 1,
						});
```

File: src/DeckGlMap/index.js [hunk 4]
```diff
@@ -143,30 +190,18 @@ const DeckGlMap = ({
...
+					{
+						mapKey,
+						layerKey,
+						featureKey,
+						feature,
+						x,
+						y,
+					},
+				],
+				raster: tooltipData.raster,
```
Comment (vlach1989) [inline src/DeckGlMap/index.js:203]: why raster in vectorLayerHover?

File: src/DeckGlMap/layers/WmsLayer/index.js [hunk 5]
```diff
@@ -115,15 +115,19 @@ class WmsLayer extends CompositeLayer {
 					.join('&');
 
 				const finalUrl = url + '?' + urlQueryString;
-
+				console.log('xxx_finalUrl', finalUrl);
```
Comment (vlach1989) [inline src/DeckGlMap/layers/WmsLayer/index.js:118]: remove log

**Review by vlach1989 — CHANGES_REQUESTED** [review]

**Comment by vdubr** [comment]
> 

I understand, but developer can deal with it in implementation of Tooltip.

File: src/DeckGlMap/index.js (hunk 4 above)
Comment (vdubr) [reply src/DeckGlMap/index.js:202]: It use useState, so you must also reuse unmodified values.

**Comment by vlach1989** [comment]
> > 
> 
> Regards to reducing of overlaping features. I understand, but developer can deal with it in implementation of Tooltip.

Ok

**Review by vlach1989 — APPROVED** [review]

**Pull request merged**